distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -88,19 +110,45 @@ public static Action chunkedContentWithHeaders(String responseMessage, Duration
     }
 
     /**
+     * Returns a composite action that holds the headers required for a plain text response as well
+     * as the response function itself. Unless the headers included are not wanted, this is the
+     * preferred way to set the Restito response actions.
+     * <p>
+     * Note that additional headers may also be needed for specific endpoint functionality. This
+     * method returns a composite action so that additional headers can be set alongside this action.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     * @return composite action that holds the headers required for a plain text response as well
+     * as the response function itself.
+     */
+    public static Action chunkedContentWithHeaders(String responseMessage, Duration messageDelay,
+            int numberOfFailures, Function<Void, Map<String, String>> requestHeaders) {
+        return composite(getHeaders(),
+                chunkedContent(responseMessage, messageDelay, numberOfFailures, requestHeaders));
+    }
+
+    /**
      * Private inner class representing the response function
      */
-    private static class ChunkedContentFunction implements Function<Response, Response> {
-        private String responseMessage;
+    private static class ChunkedContentFunction implements Function<Response, Response>,
```

Comment:
Do we really need to implement both `Function` _and_ `com.xebialabs.restito.semantics.Function`?